app/models/metric/ci_mixin/capture.rb

```diff
@@ -17,6 +17,20 @@ def queue_name_for_metrics_collection
     ems.metrics_collector_queue_name
   end
 
+  def split_capture_intervals(start_time, end_time)
+    intervals = []
+    start_hour = start_time
+    while start_hour != end_time
+      #TODO: replace 1.day with a configurable threshold defined by provider type
+      end_hour = start_hour + 1.day
+      end_hour = end_time if end_hour > end_time
+      intervals << [start_hour, end_hour]
+      start_hour = end_hour
+    end
```

Comment:
This also feels very "copy'n'paste" :smile: 

I can update it to something prettier.